synapse/storage/devices.py

```diff
@@ -814,6 +835,9 @@ def _add_device_change_txn(self, txn, user_id, device_ids, hosts, stream_id):
             ],
         )
 
+        context = {"opentracing": {}}
+        opentracing.inject_active_span_text_map(context["opentracing"])
```

Comment:
this feels like a confusing way to write this.

Could it look like this instead:

```
context = {
    "opentracing": opentracing.get_active_span_text_map()
}
```

Also: shouldn't we be passing the `destination` into `inject_active_span_text_map` ?

Reply:
I've added a method for that.

Destination is checked just below it on a per message basis.